sleap_io/io/nwb.py

```diff
@@ -14,21 +14,131 @@
+from pynwb.image import ImageSeries
+from pynwb.testing.mock.utils import name_generator
+
+from ndx_pose import (
+    PoseEstimationSeries,
+    PoseEstimation,
+    Skeleton as NWBSkeleton,
+    SkeletonInstance,
+    SkeletonInstances,
+    TrainingFrame,
+    TrainingFrames,
+    PoseTraining,
+    SourceVideos,
+)
 
 from sleap_io import (
     Labels,
     Video,
     LabeledFrame,
     Track,
-    Skeleton,
+    Skeleton as SLEAPSkeleton,
     Instance,
     PredictedInstance,
 )
 from sleap_io.io.utils import convert_predictions_to_dataframe
 
 
+def pose_training_to_labels(pose_training: PoseTraining) -> Labels:  # type: ignore[return]
+    """Creates a Labels object from an NWB PoseTraining object."""
+    labeled_frames = []
+    for training_frame in pose_training.training_frames:
+        video = Video(filename=training_frame.source_video)
+        frame_idx = training_frame.source_video_frame_index
+        instances = [
+            Instance.from_numpy(
+                points=inst,
+                point_scores=np.ones(inst.shape[0]),
+                instance_score=1.0,
+                skeleton=NWBSkeleton(
+                    nodes=pose_training.skeleton.nodes,
+                    edges=pose_training.skeleton.edges,
+                ),
+            )
+            for inst in training_frame.skeleton_instances
+        ]
+        labeled_frames.append(
+            LabeledFrame(video=video, frame_idx=frame_idx, instances=instances)
+        )
+    return Labels(labeled_frames)
+
+
+def labels_to_pose_training(labels: Labels, filename: str, **kwargs) -> PoseTraining:  # type: ignore[return]
+    """Creates an NWB PoseTraining object from a Labels object."""
+    training_frame_list = []
+    for i, labeled_frame in enumerate(labels.labeled_frames):
+        training_frame_name = name_generator("training_frame")
+        training_frame_annotator = f"{training_frame_name}{i}"
+        training_frame_skeleton_instances = SkeletonInstances(
+            [
+                instance_to_skeleton_instance(instance)
+                for instance in labeled_frame.instances
+            ]
+        )
+        training_frame_video = labeled_frame.video
+        training_frame_video_index = labeled_frame.frame_idx
+        training_frame = TrainingFrame(
+            name=training_frame_name,
+            annotator=training_frame_annotator,
+            skeleton_instances=training_frame_skeleton_instances,
+            source_video=training_frame_video.filename,
+            source_video_frame_index=training_frame_video_index,
+        )
+        training_frame_list.append(training_frame)
+    training_frames = TrainingFrames(training_frames=training_frame_list)
+
+    source_videos = labels.videos[0].filename
+    pose_training = PoseTraining(
+        training_frames=training_frames, source_videos=source_videos
+    )
+    return pose_training
+
+def slp_skeleton_to_nwb(skeleton: SLEAPSkeleton) -> NWBSkeleton: # type: ignore[return]
```

Comment:
`NWBSkeleton` and `SLEAPSkeleton` are type aliases that I wrote because both classes are named `Skeleton`, which can lead to some confusion.
https://github.com/keyaloding/io_fork/blob/f6d621ced48d62a09e126a5f5bb5e8237de6b291/sleap_io/io/nwb.py#L25
https://github.com/keyaloding/io_fork/blob/f6d621ced48d62a09e126a5f5bb5e8237de6b291/sleap_io/io/nwb.py#L39